### PR TITLE
Sanitize metrics

### PR DESCRIPTION
**Pull request by ericchu94**
gh-33

### CODE REVIEW

File: src/fullerite/metric/metric_test.go [hunk 1]
```diff
@@ -64,3 +64,133 @@ func TestGetDimensionValueNotFound(t *testing.T) {
...
+	assert.Equal(t, "DirtyMetric-", m.Name, "metric name should be sanitized")
+}
+
+func TestSanitizeMetricNameEqual(t *testing.T) {
+	m := metric.New("DirtyMetric=")
+	assert.Equal(t, "DirtyMetric-", m.Name, "metric name should be sanitized")
+}
+
+func TestSanitizeDimensionNameColon(t *testing.T) {
+	defaultDimensions := make(map[string]string, 0)
```
Comment (squirrely) [inline src/fullerite/metric/metric_test.go]: make(map[string]string)

File: src/fullerite/metric/metric_test.go [hunk 2]
```diff
@@ -64,3 +64,133 @@ func TestGetDimensionValueNotFound(t *testing.T) {
...
+	m := metric.New("TestMetric")
+	m.AddDimension("DirtyDimension:", "dimension value")
+	assert := assert.New(t)
+
+	value, ok := m.GetDimensionValue("DirtyDimension:", defaultDimensions)
+	assert.Equal("dimension value", value, "dimension value does not match")
+	assert.True(ok, "should succeed")
```
Comment (squirrely) [inline src/fullerite/metric/metric_test.go]: these comments aren't really helpful. Testing true here is enough context.

File: src/fullerite/metric/metric_test.go (hunk 1 above)
Comment (ericchu94) [inline src/fullerite/metric/metric_test.go]: done

File: src/fullerite/metric/metric_test.go (hunk 2 above)
Comment (ericchu94) [inline src/fullerite/metric/metric_test.go]: removed all comments for assert.True and assert.False

**Comment by ericchu94** [comment]
I added the overwriting dimensions tests, and also internal state tests

**Comment by baris** [comment]
Looks very good.

**Pull request merged**